### PR TITLE
UI-243 : Added rich text component with quill

### PR DESCRIPTION
**Pull request by GhitaNektt**
## JIRA Ticket

[UI-243](https://youcanshop.atlassian.net/browse/UI-243)

## QA Steps

* [ ] Go to X and Y
* [ ] See TEXT in page Z

## Note

Leave empty when you have nothing to say

[UI-243]: https://youcanshop.atlassian.net/browse/UI-243?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/vue3/src/components/RichText/EditorToolbar.ts
Comment (eihabkhan1) [inline packages/vue3/src/components/RichText/EditorToolbar.ts:1]: Seems like we are missing the `x-100` color degree from each color (e.g `brand-100`, `gray-100`, etc)

File: packages/vue3/src/components/RichText/ToolbarIcons.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    'justify': '<i class="i-youcan:text-align-justify"></i>',
+  },
+  'background': '<i class="i-youcan:paint-bucket"></i>',
+  'blockquote': '<i class="i-youcan:truck"></i>',
+  'bold': '<i class="i-youcan:text-bolder"></i>',
+  'clean': '<i class="i-youcan:truck"></i>',
+  'code': '<i class="i-youcan:code"></i>',
+  'code-block': '<i class="i-youcan:code"></i>',
+  'color': '<i class="i-youcan:eyedropper-sample"></i>',
+  'direction': {
```
Comment (eihabkhan1) [inline packages/vue3/src/components/RichText/ToolbarIcons.ts:15]: We should add a different icon that represents RTL/LTR

**Comment by eihabkhan1** [comment]
Issues I found while using RichText:
- Toolbar is not sticky, when the content is long, the toolbar doesn't stick
- Italic style only working when the text is bold
- "T" icon should change to something that represents 
- `webp` and `avif` images are not supported
- Can only upload videos from a link, can't upload a video from local machine

**Review by eihabkhan1 — CHANGES_REQUESTED** [review]

**Comment by eihabkhan1** [comment]
Seems that we can't delete uploaded images

**Review by mosarabi — CHANGES_REQUESTED** [review]
https://github.com/youcan-shop/youcan-ui/assets/149577299/2bcb473a-e86f-48ac-a55d-926e97fb7901

The Quill Editor block does not function correctly in RTL mode

**Comment by GhitaNektt** [comment]
Try configuring the direction in the code prior to mounting the component.

**Review by mosarabi — APPROVED** [review]

**Pull request merged**